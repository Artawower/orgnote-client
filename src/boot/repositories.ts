import { boot } from 'quasar/wrappers';
import { Database, NoteRepository, FileRepository } from '../repositories';

const db = new Database(NoteRepository, FileRepository);

const repositories = {
  notes: new NoteRepository(db),
  files: new FileRepository(db),
} as const;

export default boot(({ app }) => {
  app.config.globalProperties.$db = db;
  app.config.globalProperties.$repositories = repositories;
});

export { db, repositories };
