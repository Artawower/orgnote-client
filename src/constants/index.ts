export * from './initial-note-template';
export * from './image-file-extensions';
