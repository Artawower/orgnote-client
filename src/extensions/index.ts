import type { ExtensionManifest, Extension, ActiveExtension } from 'orgnote-api';

// TODO: feat/stable-beta delete
// eslint-disable-next-line @typescript-eslint/no-unused-vars
const registerActiveExtension = (manifest: ExtensionManifest, ext: Extension) => ({
  [manifest.name]: {
    manifest,
    module: ext,
    active: true,
  },
});

export const BUILTIN_EXTENSIONS: { [key: string]: ActiveExtension } = {};
