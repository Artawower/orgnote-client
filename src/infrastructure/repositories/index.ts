import type { OrgNoteApi } from 'orgnote-api';
import { createDatabase } from './create-database';
import { NOTE_REPOSITORY_NAME, NOTE_MIGRATIONS } from './note-info-repository';
import { FILE_REPOSITORY_NAME, FILE_MIGRATIONS } from './file-repository';
import { createFileRepository } from './file-repository';
import { createNoteInfoRepository } from './note-info-repository';

export async function initRepositories(): Promise<OrgNoteApi['infrastructure']> {
  const { db } = createDatabase([
    { storeName: NOTE_REPOSITORY_NAME, migrations: NOTE_MIGRATIONS },
    { storeName: FILE_REPOSITORY_NAME, migrations: FILE_MIGRATIONS },
  ]);
  return {
    fileInfoRepository: createFileRepository(db),
    noteInfoRepository: createNoteInfoRepository(db),
  };
}
