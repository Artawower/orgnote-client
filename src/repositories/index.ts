export * from './note-repository';
export * from './repository';
export * from './database';
export * from './file-repository';
export * from './orgnote-database';
