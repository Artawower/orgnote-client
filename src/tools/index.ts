export * from './extract-file';
export * from './quill-formatter';
export * from './tinykeys-wrapper';
export * from './quill-blots';
export * from './generate-file-name';
export * from './remove-file-extension';
export * from './debounce';
export * from './is-org-file';
export * from './read-file';
export * from './is-url';
export * from './get-file-name';
export * from './sleep';
