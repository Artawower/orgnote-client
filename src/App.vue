<template>
  <router-view />
</template>

<script lang="ts">
export default {
  name: 'App',
};
</script>

<script lang="ts" setup>
import { useRouter } from 'vue-router';
import { jsBabel } from './plugins/js-babel';
import { RouteNames } from './router/routes';
import {
  useAuthStore,
  useOrgBabelStore,
  useLoggerStore,
  useSyncStore,
  useOrgNoteApiStore,
} from './stores';
import { decodeAuthState, extractAuthQueryInfo } from './tools';

const syncStore = useSyncStore();

useLoggerStore().init();

syncStore.markToSync();

const orgBabelStore = useOrgBabelStore();
orgBabelStore.register(jsBabel);

if (window.navigator.standalone) {
  document.body.classList.add('standalone');
}

const authStore = useAuthStore();

const router = useRouter();

// TODO: master move to external method.
async function handleCordovaAuth(url: string) {
  const urlParams = url.split('?')?.[1];
  if (!urlParams) {
    return;
  }
  const searchParams = new URLSearchParams(urlParams);

  const personalInfo = extractAuthQueryInfo(Object.fromEntries(searchParams));
  console.debug('✎: [AUTH] personalInfo: ', personalInfo);
  const state = decodeAuthState(searchParams.get('state'));
  console.debug('✎: [AUTH] state: ', state);

  await authStore.authUser(personalInfo, searchParams.get('token'));

  if (state.redirectUrl) {
    window.location.assign(`/#${state.redirectUrl}`);
    return;
  }

  router.push({ name: RouteNames.Home });
}
(window as unknown as { handleOpenURL: (arg0: string) => void }).handleOpenURL =
  handleCordovaAuth.bind(this);

const { orgNoteApi } = useOrgNoteApiStore();
window.orgnote = orgNoteApi;
</script>
