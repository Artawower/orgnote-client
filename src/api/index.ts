import { Note } from 'src/models';

export interface OrgNoteApi {
  navigation: {
    openNote: (id: string) => void;
    editNote: (id: string) => void;
  };
  currentNote: {
    getCurrentNote: () => Note;
  };
  configuration: () => OrgNoteConfig;
}

// NOTE: default configs
export interface OrgNoteConfig {
  editor: {
    showSpecialChars: boolean;
  };
}
