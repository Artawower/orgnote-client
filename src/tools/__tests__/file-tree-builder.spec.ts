import { FileTree, FileNode } from 'src/repositories';
import {
  convertFlatTreeToFileTree,
  FlatTree,
  mergeFilesTrees,
  renameFileInTree,
} from '../file-tree-builder';

describe('file-tree-builder', () => {
  it('Should merge two file tree with preserving folders', () => {
    const srcTree: FileTree = {
      d1: {
        filePath: [],
        name: 'd1',
        type: 'folder',
        children: {},
      },
    };

    const dstTree: FileTree = {
      f1: {
        filePath: [],
        name: 'f1',
        type: 'file',
      },
    };

    const mergedTree = mergeFilesTrees(srcTree, dstTree);

    // TODO: master inline snapshots are not supported yet https://bun.sh/guides/test/snapshot
    expect(mergedTree).toMatchSnapshot();
  });

  it('Should merge two file tree with preserving nested folders', () => {
    const srcTree: FileTree = {
      d1: {
        filePath: [],
        name: 'd1',
        type: 'folder',
        children: {
          d2: {
            filePath: ['d1'],
            name: 'd2',
            type: 'folder',
            children: {},
          },
        },
      },
    };

    const dstTree: FileTree = {
      f1: {
        filePath: [],
        name: 'f1',
        type: 'file',
      },
    };

    const mergedTree = mergeFilesTrees(srcTree, dstTree);
    expect(mergedTree).toMatchSnapshot();
  });

  it('Should remove files from src directory if they are not in dst', () => {
    const srcTree: FileTree = {
      d1: {
        filePath: [],
        name: 'd1',
        type: 'folder',
        children: {
          f1: {
            filePath: ['d1'],
            name: 'f1',
            type: 'file',
          },
          f2: {
            filePath: ['d1'],
            name: 'f2',
            type: 'file',
          },
        },
      },
    };

    const dstTree: FileTree = {
      d1: {
        filePath: [],
        name: 'd1',
        type: 'folder',
        children: {},
      },
    };

    const mergedTree = mergeFilesTrees(srcTree, dstTree);
    expect(mergedTree).toMatchSnapshot();
  });

  it('Should rename nested file', () => {
    const t: FileTree = {
      d1: {
        filePath: [],
        name: 'd1',
        type: 'folder',
        children: {
          f1: {
            filePath: ['d1'],
            name: 'f1',
            id: 'id',
            type: 'file',
          },
        },
      },
    };

    const fileNode: FileNode = {
      name: 'f1',
      id: 'id',
      filePath: ['d1'],
      type: 'file',
    };

    const newName = 'newName';

    const updatedTree = renameFileInTree(t, fileNode, newName);

    expect(updatedTree).toMatchSnapshot();
  });

  it('Should rename nested files path', () => {
    const t: FileTree = {
      d1: {
        filePath: [],
        name: 'd1',
        type: 'folder',
        children: {
          d2: {
            filePath: ['d1'],
            name: 'd2',
            type: 'folder',
            children: {
              f1: {
                filePath: ['d1', 'd2'],
                id: 'id',
                name: 'f1',
                type: 'file',
              },
            },
          },
        },
      },
    };

    const fileNode: FileNode = {
      name: 'd2',
      filePath: ['d1'],
      type: 'folder',
    };

    const newFolderName = 'newFolderName';
    const newTree = renameFileInTree(t, fileNode, newFolderName);
    expect(newTree).toMatchSnapshot();
  });

  it('Should convert flat tree to file tree', () => {
    const flatTree: FlatTree = {
      name: 'd1',
      type: 'folder',
      filePath: [],
      children: [
        {
          name: 'd2',
          type: 'folder',
          filePath: ['d1'],
          children: [
            {
              name: 'f1',
              type: 'file',
              id: 'fileId',
              filePath: ['d1', 'd2'],
            },
          ],
        },
        {
          name: 'f2',
          type: 'file',
          id: 'fileId',
          filePath: ['d1'],
        },
      ],
    };

    const fileTree = convertFlatTreeToFileTree(flatTree);
    expect(fileTree).toMatchSnapshot();
  });
});
