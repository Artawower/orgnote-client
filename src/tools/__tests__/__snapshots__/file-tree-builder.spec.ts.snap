// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`file-tree-builder Should merge two file tree with preserving folders 1`] = `
{
  "d1": {
    "children": {},
    "filePath": [],
    "name": "d1",
    "type": "folder",
  },
  "f1": {
    "filePath": [],
    "name": "f1",
    "type": "file",
  },
}
`;

exports[`file-tree-builder Should merge two file tree with preserving nested folders 1`] = `
{
  "d1": {
    "children": {
      "d2": {
        "children": {},
        "filePath": [
          "d1",
        ],
        "name": "d2",
        "type": "folder",
      },
    },
    "filePath": [],
    "name": "d1",
    "type": "folder",
  },
  "f1": {
    "filePath": [],
    "name": "f1",
    "type": "file",
  },
}
`;

exports[`file-tree-builder Should remove files from src directory if they are not in dst 1`] = `
{
  "d1": {
    "children": {},
    "filePath": [],
    "id": "d1",
    "name": "d1",
    "type": "folder",
  },
}
`;

exports[`file-tree-builder Should rename nested file 1`] = `
[
  {
    "d1": {
      "children": {
        "newName": {
          "filePath": [
            "d1",
          ],
          "id": "id",
          "name": "newName",
          "type": "file",
        },
      },
      "filePath": [],
      "name": "d1",
      "type": "folder",
    },
  },
  [
    {
      "filePath": [
        "d1",
        "newName",
      ],
      "id": "id",
    },
  ],
]
`;

exports[`file-tree-builder Should rename nested files path 1`] = `
[
  {
    "d1": {
      "children": {
        "newFolderName": {
          "children": {
            "f1": {
              "filePath": [
                "d1",
                "newFolderName",
              ],
              "id": "id",
              "name": "f1",
              "type": "file",
            },
          },
          "filePath": [
            "d1",
          ],
          "name": "newFolderName",
          "type": "folder",
        },
      },
      "filePath": [],
      "name": "d1",
      "type": "folder",
    },
  },
  [
    {
      "filePath": [
        "d1",
        "newFolderName",
        "f1",
      ],
      "id": "id",
    },
  ],
]
`;

exports[`file-tree-builder Should convert flat tree to file tree 1`] = `
{
  "children": {
    "d2": {
      "children": {
        "f1": {
          "children": {},
          "filePath": [
            "d1",
            "d2",
          ],
          "id": "fileId",
          "name": "f1",
          "type": "file",
        },
      },
      "filePath": [
        "d1",
      ],
      "id": undefined,
      "name": "d2",
      "type": "folder",
    },
    "f2": {
      "children": {},
      "filePath": [
        "d1",
      ],
      "id": "fileId",
      "name": "f2",
      "type": "file",
    },
  },
  "filePath": [],
  "id": undefined,
  "name": "d1",
  "type": "folder",
}
`;

exports[`file-tree-builder Should to have initial id after merging 1`] = `
{
  "d1": {
    "children": {
      "f1": {
        "filePath": [
          "d1",
        ],
        "id": "id2",
        "name": "f1",
        "type": "file",
      },
    },
    "filePath": [],
    "id": "id1",
    "name": "d1",
    "type": "folder",
  },
}
`;

exports[`file-tree-builder Should to prefer source tree folder id over dst tree with same name 1`] = `
{
  "d1": {
    "children": {
      "f1": {
        "filePath": [
          "d1",
        ],
        "id": "id3",
        "name": "f1",
        "type": "file",
      },
    },
    "filePath": [],
    "id": "id1",
    "name": "d1",
    "type": "folder",
  },
}
`;
